akka-stream/src/main/scala/akka/stream/scaladsl/Framing.scala

```diff
@@ -193,7 +193,7 @@ object Framing {
         } else if (buffer.slice(possibleMatchPos, possibleMatchPos + separatorBytes.size) == separatorBytes) {
           // Found a match
           val parsedFrame = buffer.slice(0, possibleMatchPos).compact
-          buffer = buffer.drop(possibleMatchPos + separatorBytes.size)
+          buffer = buffer.drop(possibleMatchPos + separatorBytes.size).compact
```

Comment:
I don't think it is needed. I used to use the heuristic that I record the position of the last compaction, and then compact once the position of the first byte of the buffer is larger than the last position (i.e. there is no overlap anymore with the original ByteString after the last compaction).